src/main/java/step1/StringCalculator.java

```diff
@@ -0,0 +1,56 @@
+package step1;
+
+
+public class StringCalculator {
+
+    private final Numbers numbers;
+    private final Operators operators;
+    private int result;
+
+
+    public StringCalculator(String string) {
+        validateNull(string);
+        this.numbers = new Numbers(string);
+        this.operators = new Operators(string);
+        this.result = 0;
+    }
+
+
+    private static void validateNull(String string) {
+        if (string == null || string.isBlank()) {
+            throw new IllegalArgumentException("문자열이 빈값일 수 없습니다.");
+        }
+    }
+
+    public int calculate() {
+        int index = 0;
+        result = numbers.getNumbers(index);
+        for (Symbol symbols : operators.getSymbols()) {
+            calculatedWith(symbols, numbers.getNumbers(++index));
+        }
```

Comment:
```suggestion
        for (Symbol symbols : operators.getSymbols()) {
            result = symbol.calculate(result, numbers.getNumbers(++index));
        }
```
추상화가 잘 되었다면 위와 같은 코드로 계산을 돌려볼 수 있겠네요 😄